### PR TITLE
Alps support distributed running

### PR DESCRIPTION
**Pull request by joyyoj**
user can set worker num and ps num
Currently, support follows attrs:
* engine.worker_num
* engine.worker_memory
* engine.ps_num
* engine.ps_memory

```sql
...
train dnnclassifier 
with model.n_classes = 2, engine.worker_num = 2, engine.ps_num = 1
```
If worker_num > 1, it will run distributed training.

### CODE REVIEW

**Comment by typhoonzero** [comment]
Please try resolve conflict by running `git pull upstream develop`, thanks

File: sql/expression_resolver.go [hunk 1]
```diff
@@ -128,152 +141,114 @@ type embeddingColumn struct {
...
+	if p, ok := attrs["type"]; ok {
+		strVal, ok := p.Value.(string)
+		if ok {
+			engineType = strVal
 		}
-		delete(attrs, p.FullName)
+	} else if psNum > 0 || workerNum > 0 {
+		engineType = "k8s"
 	}
+	return engineSpec{
```
Comment (typhoonzero) [inline sql/expression_resolver.go:169]: Is there a way to let the submitter just return the yarn job URL and exit? then the status can be fetched on the web page and do not wait on the sqlflow client (e.g. jupyter notebook)

Comment (joyyoj) [reply sql/expression_resolver.go:169]: I can modify to exit immediately, but how the web page can get the status?

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**